src/tests/Regressions/coreclr/GitHub_84694/test84694.cs

```diff
@@ -0,0 +1,26 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+public class C0
+{
+}
+
+public struct S0
+{
+    public C0 F0;
+    public C0 F1;
+    public S0(C0 f1) : this()
+    {
+    }
+}
+
+public class Program
+{
+    public static int Main()
+    {
+        var vr4 = new S0[,] { { new S0(new C0()) } };
```

Comment:
```suggestion
        GC.KeepAlive(new S0[,] { { new S0(new C0()) } });
```
to make sure that the JIT won't optimize out the allocation